profile/src/main/java/rdx/works/profile/domain/GetProfileUseCase.kt

```diff
@@ -126,16 +125,10 @@ suspend fun GetProfileUseCase.accountOnCurrentNetwork(
     account.address == withAddress
 }
 
-suspend fun GetProfileUseCase.nextDerivationPathForAccountOnNetwork(
-    derivationPathScheme: DerivationPathScheme,
-    networkId: Int,
-    factorSourceId: FactorSource.FactorSourceID
-): DerivationPath {
-    val profile = invoke().first()
-    val network = requireNotNull(NetworkId.from(networkId))
+fun Network.nextDerivationPathForAccountOnNetwork(factorSource: FactorSource): DerivationPath {
```

Comment:
maybe we should also create NetworkExtensions.kt and move all functions that operate on Network there.

Reply:
Yes I agree, I would also argue that most of the "extensions" on `GetProfileUseCase` should be extensions on `Profile` and not the usecase itself. So we would end up having ProfileExtensions and NetworkExtensions